middle_end/flambda/inlining/inlining_transforms.ml

```diff
@@ -23,6 +23,169 @@ module DE = Downwards_env
 module I = Flambda_type.Function_declaration_type.Inlinable
 module VB = Var_in_binding_pos
 
+let make_inlined_body ~callee ~unroll_to ~params ~args ~my_closure ~body
+      ~exn_continuation ~return_continuation ~apply_exn_continuation
+      ~apply_return_continuation =
+  let perm = Name_permutation.empty in
+  let perm =
+    match (apply_return_continuation : Apply.Result_continuation.t)with
+    | Return k ->
+       Name_permutation.add_continuation perm
+         return_continuation k
+    | Never_returns -> perm
+  in
+  let perm =
+    Name_permutation.add_continuation perm
+      (Exn_continuation.exn_handler exn_continuation)
+      apply_exn_continuation
+  in
+  let callee =
+    Simple.merge_rec_info callee
+      ~newer_rec_info:(Some (Rec_info.create ~depth:1 ~unroll_to))
+    |> Option.get  (* CR mshinwell: improve *)
+  in
+  Expr.apply_name_permutation
+    (Expr.bind_parameters_to_args_no_simplification
+       ~params ~args
+       ~body:(Let.create
+                (Bindable_let_bound.singleton
+                   (VB.create my_closure Name_mode.normal))
+                (Named.create_simple callee)
+                ~body
+                (* Here and below, we don't need to give any name occurrence
+                      information (thank goodness!) since the entirety of the
```

Comment:
Indentation of comment has gone wrong here

Reply:
Done